lib/cuda/cudaOutputData.cpp

```diff
@@ -12,134 +12,123 @@ using kotekan::Config;
 REGISTER_CUDA_COMMAND(cudaOutputData);
 
 cudaOutputData::cudaOutputData(Config& config, const std::string& unique_name,
-                               bufferContainer& host_buffers, cudaDeviceInterface& device) :
-    cudaCommand(config, unique_name, host_buffers, device, "", "") {
+                               bufferContainer& host_buffers, cudaDeviceInterface& device,
+                               int instance_num) :
+    cudaCommand(config, unique_name, host_buffers, device, instance_num) {
     std::string in_buf_name = config.get_default<std::string>(unique_name, "in_buf", "");
     if (in_buf_name.size()) {
         in_buffer = host_buffers.get_buffer(in_buf_name);
-        register_consumer(in_buffer, unique_name.c_str());
+        if (instance_num == 0)
+            register_consumer(in_buffer, unique_name.c_str());
     } else
         in_buffer = nullptr;
 
     output_buffer = host_buffers.get_buffer(config.get<std::string>(unique_name, "out_buf"));
-    register_producer(output_buffer, unique_name.c_str());
-
-    for (int i = 0; i < output_buffer->num_frames; i++) {
-        uint flags;
-        // only register the memory if it isn't already...
-        if (cudaErrorInvalidValue == cudaHostGetFlags(&flags, output_buffer->frames[i])) {
-            CHECK_CUDA_ERROR(
-                cudaHostRegister(output_buffer->frames[i], output_buffer->frame_size, 0));
+    if (instance_num == 0) {
+        register_producer(output_buffer, unique_name.c_str());
+        for (int i = 0; i < output_buffer->num_frames; i++) {
+            uint flags;
+            // only register the memory if it isn't already...
+            if (cudaErrorInvalidValue == cudaHostGetFlags(&flags, output_buffer->frames[i])) {
+                CHECK_CUDA_ERROR(
+                    cudaHostRegister(output_buffer->frames[i], output_buffer->frame_size, 0));
+            }
```

Comment:
Not really a part of this PR, so feel free to leave for another time, but I think we can actually get rid of this logic.   It should be handled by gpuProcess.cpp:97 for all buffers.

Reply:
Done!